### PR TITLE
switch from pyparsing to lark

### PR DESCRIPTION
**Pull request by Roguelazer**
Switch from PyParsing combinators to a LALR parser using [lark](https://github.com/lark-parser/lark).

This is approximately 3x faster on my hardware (~700µs to ~240µs)

### CODE REVIEW

**Comment by pvinci** [comment]
Would you be willing to add this message as a test under lark?
message: <134>1 2019-01-20T23:43:41.087236Z 172.16.3.1 NAT 15634 SADD [nsess SSUBIX="0" SVLAN="0" IATYP="IPv4" ISADDR="172.16.1.2" ISPORT="6303" XATYP="IPv4" XSADDR="10.0.0.3" XSPORT="16253" PROTO="6" XDADDR="172.16.2.2" XDPORT="80"] <type 'str'>

File: syslog_rfc5424_parser/parser.py [hunk 1]
```diff
@@ -1,64 +1,132 @@
...
-
-
-__all__ = ['syslog_message', 'structured_data', 'header']
+import collections
+
+from lark import Lark, Transformer
+
+
+GRAMMAR = '''
+    ?start           : header _SP structured_data [ msg ]
```
Comment (jaredg) [inline syslog_rfc5424_parser/parser.py]: Does the question mark prefix even make sense on these?  Can these be inlined?

File: syslog_rfc5424_parser/parser.py [hunk 2]
```diff
@@ -1,64 +1,132 @@
...
+            rest = [str(i.children[0]) for i in inp[1:]]
+            return datetime + ''.join(rest)
+
+    def hostname(self, inp):
+        return str(inp[0])
+
+    def appname(self, inp):
+        return str(inp[0])
+
+    def procid(self, inp):
```
Comment (jaredg) [inline syslog_rfc5424_parser/parser.py]: I don't remember if it's cheaper to convert an int to a string back to an int or to test if it's an int initially.

**Comment by Roguelazer** [comment]
@pvinci done

**Comment by pvinci** [comment]
Thanks!

**Pull request merged**